code/game/turfs/simulated/wall_attacks.dm

```diff
@@ -69,27 +68,44 @@
 
 	try_touch(user, rotting)
 
-/turf/simulated/wall/attack_generic(var/mob/user, var/damage, var/attack_message, var/wallbreaker)
+/turf/simulated/wall/attack_generic(mob/M, damage, attack_message)
+	log_and_message_admins("attacked [src]: [jumplink(src)]")
+	M.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
 
-	radiate()
-	if(!istype(user))
-		return
+	if(locate(/obj/effect/overlay/wallrot) in src)
+		return success_smash(M)
 
-	user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
-	var/rotting = (locate(/obj/effect/overlay/wallrot) in src)
-	if(!damage || !wallbreaker)
-		try_touch(user, rotting)
+	radiate()
+	if(!istype(M))
 		return
 
-	if(rotting)
-		return success_smash(user)
-
 	if(reinf_material)
-		if((wallbreaker == 2) || (damage >= max(material.hardness,reinf_material.hardness)))
-			return success_smash(user)
-	else if(damage >= material.hardness)
-		return success_smash(user)
-	return fail_smash(user)
+		if (damage < max(material.hardness + reinf_material.hardness/2))
+			return attack_hand(M)
+		else
+			playsound(src, pick(WALLHIT_SOUNDS), 50, 1)
+			M.do_attack_animation(src)
+			M.visible_message(SPAN_DANGER("\The [M] [attack_message] \the [src]!"))
+			return take_damage(damage)
+
+	if(!damage || damage < material.hardness/6)
+		return attack_hand(M)
+
+	if(damage < material.hardness/2)
+		if(prob(50))
+			M.do_attack_animation(src)
+			M.visible_message(SPAN_DANGER("\The [M] [attack_message] the [src] to no effect!"))
+			return attack_hand(M)
+		else
+			playsound(src, pick(WALLHIT_SOUNDS), 50, 1)
+			M.do_attack_animation(src)
+			M.visible_message(SPAN_DANGER("\The [M] [attack_message] \the [src]!"))
+			return take_damage(damage)
+	if(damage > material.hardness/2)
+		playsound(src, pick(WALLHIT_SOUNDS), 50, 1)
+		M.do_attack_animation(src)
+		M.visible_message(SPAN_DANGER("\The [M] smashes \the [src]!"))
+		take_damage(damage)
```

Comment:
```suggestion
	var/rot = locate(/obj/effect/overlay/wallrot) in src
	var/hardness = reinf_material ? max(material.hardness, reinf_material.hardness) : material.hardness
	if(!damage)
		try_touch(user, rot)
	else if(damage >= hardness)
		return success_smash(user)
	else
		fail_smash(user)
```

Reply:
Unless I'm reading it wrong, won't fail_smash cause a low damage mob to deal 25-75 damage to a wall and if the mob matches the wall's hardness it'll break the wall instantly?

Reply:
Yeah.

Reply:
Is this an example of what the code /should look/ like, or what it's actually meant to be? This means a 2 damage roach can destroy a wall in 2 hits if it's lucky?

Reply:
This is the way that proc worked originally, it's not great balance-wise, but since roach AI can't target structures - there haven't been an issue so far.

Reply:
Is there a reason to keep it that way, though..? Not just roaches but light mechs shouldn't IMO be breaking a wall in 2-4 hits     but I guess the girder being absurdly strong helps balance that out.

Reply:
There is no such reason, change it if you want.
¯\\\_(ツ)\_/¯